test/helpers/vertical_nav_helper_test.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class VerticalNavHelperTest < ActionView::TestCase
+
+  test '#backend_api_nav_sections' do
```

Comment:
What does happen if: 

1. `@backend_api = nil`
2. `@backend_api.persisted? == false`

Reply:
correct this test is missing and I will cover this.

Reply:
I don't think it will be nil ever, because the side nav bar open when the api creation fails which is case of falseor else created successfully,
 
if @backend_api is nil then persisted? method will also fails and give nil error.

Reply:
Yes, exactly. This is unit test, meaning it is isolated from the context. So your argument is not valid in this case ;)

Even if the scenario described does not happen now, we introduce a potential error leading to a 500 to the customer :) You do not know the future, but you know that this case might happen. Again not in our case but out of context it does. So I do not think we want to take that risk.

FYI do not mistake unit testing (without context) and integration testing (with context).